src/wayland-egldisplay.c

```diff
@@ -670,48 +674,103 @@ EGLBoolean wlEglTerminateHook(EGLDisplay dpy)
     return res;
 }
 
-static void getServerProtocolsInfo(struct wl_display *nativeDpy,
+static char* getNodeFromDrmDevice(dev_t dev) {
+    char *result = NULL;
+    drmDevice *drm_device;
+    if (drmGetDeviceFromDevId(dev, 0, &drm_device) == 0) {
+        if (drm_device->available_nodes & (1 << DRM_NODE_RENDER)) {
+            result = strdup(drm_device->nodes[DRM_NODE_RENDER]);
+        } else if (drm_device->available_nodes & (1 << DRM_NODE_PRIMARY)) {
+            result = strdup(drm_device->nodes[DRM_NODE_PRIMARY]);
```

Comment:
In `wlEglGetPlatformDisplayExport` we only check if the render node of an `EGLDevice` matches `drm_name`. Therefore I don't think there's any point returning a primary node path from this function. If there's no render node available we can just return NULL.